ryot/src/directional.rs

```diff
@@ -0,0 +1,143 @@
+use std::ops::Deref;
+
+use glam::IVec2;
+use serde::{Deserialize, Serialize};
+use strum::EnumIter;
+
+#[cfg(feature = "bevy")]
+use bevy::ecs::component::Component;
+
+use crate::position::TilePosition;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy, Default, Debug)]
+#[cfg_attr(feature = "bevy", derive(Serialize, Deserialize))]
```

Comment:
We might wanna add a "serde" feature for Ser/De attributes. This is quite standard in Rust.

Reply:
We derive these basically everywhere now. I think a separate commit/PR pulling all of them into a feature would make more sense. I pulled these out of "bevy" for now